test/gui/shared/steps/steps.py

```diff
@@ -542,122 +436,62 @@ def step(context, receiver, resource):
 
 @Then('the error text "|any|" should be displayed in the sharing dialog')
 def step(context, fileShareContext):
-    test.compare(
-        str(
-            waitForObjectExists(
-                names.sharingDialog_The_file_can_not_be_shared_because_it_was_shared_without_sharing_permission_QLabel
-            ).text
-        ),
-        fileShareContext,
-    )
+    shareItem = SharingDialog()
+    shareItem.checkErrorDisplayed(fileShareContext)
```

Comment:
I think we can use the same basic rule we use on other UI tests (eg. on web). 
- do not use assertions in page objects
- page objects only return state and assertion is done in the step definitions.

Reply:
Done